File: strings-discord/src/main/java/com/pedestriamc/strings/discord/listener/bukkit/StringsReloadListener.java
package com.pedestriamc.strings.discord.listener.bukkit;

import com.pedestriamc.strings.api.event.StringsReloadEvent;
import com.pedestriamc.strings.discord.StringsDiscord;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.jetbrains.annotations.NotNull;

public class StringsReloadListener implements Listener {

    private final StringsDiscord strings;

    public StringsReloadListener(@NotNull StringsDiscord strings) {
        this.strings = strings;
    }

    @EventHandler
    void onEvent(@NotNull StringsReloadEvent event) {
        strings.reload();
    }

}

File: strings-discord/src/main/java/com/pedestriamc/strings/discord/listener/bukkit/StringsReloadListener.kt
package com.pedestriamc.strings.discord.listener.bukkit

import com.pedestriamc.strings.api.event.StringsReloadEvent
import com.pedestriamc.strings.api.event.strings.Listener
import com.pedestriamc.strings.discord.StringsDiscord
import org.jetbrains.annotations.NotNull

class StringsReloadListener(private val strings: StringsDiscord) {

    @Listener
    internal fun onEvent(@NotNull event: StringsReloadEvent) {
        strings.reload();
    }

}